soroban-env-host/src/host/conversion.rs

```diff
@@ -521,11 +518,11 @@ impl Host {
             | ScVal::I32(_)
             | ScVal::LedgerKeyNonce(_)
             | ScVal::ContractInstance(_)
-            | ScVal::LedgerKeyContractInstance => Err(err!(
-                self,
-                (ScErrorType::Value, ScErrorCode::InternalError),
+            | ScVal::LedgerKeyContractInstance => Err(self.err(
+                ScErrorType::Value,
+                ScErrorCode::InternalError,
                 "converting ScValObjRef on non-object ScVal type",
-                *val
```

Comment:
Is there a reason for not including the val in the error?

Reply:
Yeah, it's not actually a Val, it's an ScVal, which was only being handled by the `err!` macro that tries to re-convert it to Val via DebugArg. First of all this won't even work -- it's wasted work in this case as we know for sure the ScVal _isn't_ convertable to a Val -- and second of all to even make the types work requires a bunch more generic-constraint acrobatics beyond what I'm already doing in this PR, and it just didn't seem worth the effort for a thing-that-doesn't-work-anyway.